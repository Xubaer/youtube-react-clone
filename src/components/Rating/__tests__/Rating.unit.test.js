import React from 'react';
import { shallow } from 'enzyme';
import { Rating } from '../Rating';

describe('Rating', () => {
    test('renders without props', () => {
        const wrapper = shallow(<Rating />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('renders with like count and dislike count', () => {
        const wrapper = shallow(<Rating likeCount={1000} dislikeCount={50} />);
        expect(wrapper).toMatchSnapshot();
    });
});