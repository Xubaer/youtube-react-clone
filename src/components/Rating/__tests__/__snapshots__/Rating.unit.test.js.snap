// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders with like count and dislike count 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      1000
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      50
    </span>
  </div>
  <Progress
    className="progress"
    percent={95.23809523809523}
    size="tiny"
  />
</div>
`;

exports[`Rating renders without props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span />
  </div>
</div>
`;
